azerothcore/azerothcore-wotlk
# fix(Core/Player): Clear current title if the player loses it

## Changes Proposed:
-  Clear the currently chosen title if it gets removed for any reason.
-  

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game.

## How to Test the Changes:


1. Easier to be tested with mod-pvp-titles
2. get some title, set it active
3. update the kill count requirement, so you're no longer elegible to it
4. enable cleanupmode = 2 (invalid titles)
5. notice the current title is cleared

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
